### DISCUSSION TITLE
master node lost data while failover

### DISCUSSION THREAD

**Discussion by gaoshengL** [apache/kvrocks#1931] [Q&A]
I test kvrocks recently，when I kill master node，sentinel promote slave as new master.I restart the old master as slave to new master,I find that old master keys amount is less than new master.kvrocks can avoid this promble?Because of write ahead log,I think old master should recover data from binlog rather than lost data.

**Comment by git-hulk** [comment]
Hi @gaoshengL 

I don't quite understand your question. Kvrocks did recover data from the WAL of rocksdb without losing the old data. It will use the current data offset to request the increment syncing from the new master.

**Comment by gaoshengL** [comment]
Thank you for your answer。I use estimate_keys[metadata] to judge master and slave consistency，maybe it does not represent real ammount of keys

**Reply by git-hulk** [reply]
Estimate_keys includes all live and expired keys in RocksDB, so it can't be used to judge the data consistency. You can look at the value of the sequence in the INFO command, it's the WAL sequence of RocksDB and can represent the state of replication state.
